[Machine-learning][ACR-1571] Add document for new Machine Learning API (#1207)
Samsung/tizen-docs: docs/application/native/guides/machine-learning/machine-learning-inference.md

    ```c
    ml_pipeline_h handle = nullptr;
    ml_pipeline_element_h demux_h = nullptr;

    pipeline = g_strdup("videotestsrc ! video/x-raw,format=RGB,width=640,height=480 ! videorate max-rate=1 ! " \
	    "tensor_converter ! tensor_mux ! tensor_demux name=demux ! tensor_sink");

    // Construct a pipeline
    status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);

    // Get the handle of target element
    status = ml_pipeline_element_get_handle (handle, "demux", &demux_h);
    ```

    After fetching the handle of the target element, you can set and get the value of the specific property.
    ```c
    gchar *ret_tensorpick;

    // Set the string value of given element's property
    status = ml_pipeline_element_set_property_string (demux_h, "tensorpick", "1,2");

    // Get the string value of given element's property
    status = ml_pipeline_element_get_property_string (demux_h, "tensorpick", &ret_tensorpick);
    ```

>     Before you destroy the pipeline, release `demux_h` by calling `ml_pipeline_element_release_handle()`.

Reviewer: @again4you I guess you missed the comment, request you to incorporate the comment.

>     Before you destroy the pipeline, release `demux_h` by calling `ml_pipeline_element_release_handle()`.

Reviewer: @again4you Request you to please incorporate the comment. 
Once you incorporate, click on **Resolve** button.

    ```c
    ml_pipeline_element_release_handle (demux_h);
    ```

>     To figure out the property information of the target element, you can run `gst-inspect-1.0` command on your device as below.

Reviewer: @again4you I guess you missed the comment, request you to incorporate the comment.

>     To figure out the property information of the target element, you can run `gst-inspect-1.0` command on your device as below.

Reviewer: @again4you Request you to please incorporate the comment. 
Once you incorporate, click on **Resolve** button.

    ```bash
    # gst-inspect-1.0 tensor_demux
    ...
    Element Properties:
    name                : The name of the object
                            flags: readable, writable
                            String. Default: "tensordemux0"
    parent              : The parent of the object
                            flags: readable, writable
                            Object of type "GstObject"
    silent              : Produce verbose output ?
                            flags: readable, writable
                            Boolean. Default: true

>     tensorpick          : Choose nth tensor among tensors ?

Reviewer: - Why is there a? mark 

   Produces verbose output.

Also, can you please format text under element properties?

Author: Well, that is a just help message when running `gst-inspect-1.0 tensor_demux` command.
I don't know why the owner adds the question mark but it is not necessary anyway.
Since this filter is made by us so I will update the help message ASAP and remove it on this page.

    ```c
    /* Define Custom Filter function */
    static int custom_filter_invoke_cb (const ml_tensors_data_h in, ml_tensors_data_h out, void *user_data)
    {
        /* Get input tensors using data handle 'in',
           and fill output tensors using data handle 'out'. */
        if (user_data) {
            void *raw_data = NULL;
            size_t *data_size = (size_t *) user_data;

            ml_tensors_data_get_tensor_data (out, 0, &raw_data, data_size);
        }

        return 0;
    }
    ...

    ml_tensors_info_h in_info, out_info;
    ml_custom_easy_filter_h custom;

    /* Define input and output tensors information. */
    ml_tensors_info_create (&in_info);
    ml_tensors_info_set_count (in_info, 1);
    ml_tensors_info_set_tensor_type (in_info, 0, ML_TENSOR_TYPE_INT8);
    ml_tensors_info_set_tensor_dimension (in_info, 0, dim);

    ml_tensors_info_create (&out_info);
    ml_tensors_info_set_count (out_info, 1);
    ml_tensors_info_set_tensor_type (out_info, 0, ML_TENSOR_TYPE_FLOAT32);
    ml_tensors_info_set_tensor_dimension (out_info, 0, dim);

    /* Register custom filter with name 'my-custom-filter' */

>     status = ml_pipeline_custom_easy_filter_register ("my-custom-filter", in_info, out_info, custom_filter_invoke_cb, NULL, &custom);

Reviewer 2: Instead of NULL, you need to provide ```((void *) &data_size)```, where ```size_t data_size = some value or get the size with ml-tensors_info_get_tensor_size()```

Author: ACK!. My falut!

    ml_tensors_data_set_tensor_data (data, 0, dummy, 1);
    ```

## Custom Filter

> For developers' convenience, NNStreamer provides an interface for processing the tensors data with `custom-easy` framework. After registering the user-defined callback function with input/output tensors information, it can manipulate tensors data in the pipeline without an independent shared object. Since callback function works as _filter_ in the pipeline, it is named as `Custom Filter`.

Reviewer: @again4you I guess you missed the comment, request you to incorporate the comment.

> For developers' convenience, NNStreamer provides an interface for processing the tensors data with `custom-easy` framework. After registering the user-defined callback function with input/output tensors information, it can manipulate tensors data in the pipeline without an independent shared object. Since callback function works as _filter_ in the pipeline, it is named as `Custom Filter`.

Author: ACK!

> For developers' convenience, NNStreamer provides an interface for processing the tensors data with `custom-easy` framework. After registering the user-defined callback function with input/output tensors information, it can manipulate tensors data in the pipeline without an independent shared object. Since callback function works as _filter_ in the pipeline, it is named as `Custom Filter`.

Reviewer: @again4you Request you to please incorporate the comment. 
Once you incorporate, click on **Resolve** button.

- **Define and register Custom Filter**

>     Before using the Custom Filter in the pipeline, you have to register it with input/output tensors information and its name.

Reviewer: We need to replace / with appropriate word: 

Doubt: 
What is it?   
input **or** output 
input **and** output 

What does "**its**" refer to? 
 ....its name ..

The following is just a suggestion, but kindly clear my doubts to help me understand better: 

```suggestion    
    Before using the Custom Filter in the pipeline, you have to register the Custom Filter with input and output tensor information and its name.
```
If the suggestion is fine, please verify for technical correctness

Author: `Custom Filter` is a kind of user-defined function which processes the tensors data in the pipeline. To use this, input __and__ output tensors information is required. Moreover, `Custom Filter` is internally managed by its name so a unique name is needed when registering the Custom Filter.

In my opinion, your suggestion is good for me.

>     Before using the Custom Filter in the pipeline, you have to register the Custom Filter with input and output tensor information and its name.

Reviewer: ```suggestion
Before you use the Custom Filter in the pipeline, you have to register the Custom Filter with input and output tensor information and its name:

Reviewer: @again4you 

what does "its" refer to?

its = Custom Filter ??

    ```c
    /* Define Custom Filter function */
    static int custom_filter_invoke_cb (const ml_tensors_data_h in, ml_tensors_data_h out, void *user_data)
    {
        /* Get input tensors using data handle 'in',
           and fill output tensors using data handle 'out'. */
        if (user_data) {
            void *raw_data = NULL;
            size_t *data_size = (size_t *) user_data;

            ml_tensors_data_get_tensor_data (out, 0, &raw_data, data_size);
        }

        return 0;
    }
    ...

    ml_tensors_info_h in_info, out_info;
    ml_custom_easy_filter_h custom;
    size_t data_size;

>     /* Define input and output tensors information. */

Reviewer: ```suggestion
    /* Define input and output tensor information. */

    ml_tensors_info_create (&in_info);
    ml_tensors_info_set_count (in_info, 1);
    ml_tensors_info_set_tensor_type (in_info, 0, ML_TENSOR_TYPE_INT8);

>     ml_tensors_info_set_tensor_dimension (in_info, 0, dim);

Author: My falult. I will fix it.

    ml_tensors_info_create (&out_info);
    ml_tensors_info_set_count (out_info, 1);
    ml_tensors_info_set_tensor_type (out_info, 0, ML_TENSOR_TYPE_FLOAT32);
    ml_tensors_info_set_tensor_dimension (out_info, 0, dim);

    /* Register custom filter with name 'my-custom-filter' */
    status = ml_pipeline_custom_easy_filter_register ("my-custom-filter", in_info, out_info, custom_filter_invoke_cb, &data_size, &custom);
    ```

- **Construct a pipeline with Custom Filter**

>     After registering the custom filter, you can use it when constructing the pipeline.

Reviewer: C of custom & F of Filter needs to be capitalized to maintain consistency 
Suggest using uniform naming for it throughout the page. 
```suggestion 
    After registering the Custom Filter, you can use it when constructing the pipeline.

>     After registering the Custom Filter, you can use it when constructing the pipeline.

Reviewer: @again4you 

```suggestion 
 After registering the Custom Filter, you can use it when constructing the pipeline:

    ```c
    ml_pipeline_h pipe;

    /* framework is 'custom-easy' and registered model `my-custom-filter` is used */
    const char pipeline[] = "appsrc ! other/tensor,dimension=(string)2:1:1:1,type=(string)  int8,framerate=(fraction)0/1 ! tensor_filter framework=custom-easy model=my-custom-filter ! tensor_sink";

    status = ml_pipeline_construct (pipeline, NULL, NULL, &pipe);
    ```

>     After using the custom filter handle, it should be unregistered by calling `ml_pipeline_custom_easy_filter_unregister()`.

Reviewer: ```suggestion
     After using the Custom Filter handle, it should be unregistered by calling `ml_pipeline_custom_easy_filter_unregister()`:
